src/berry_pots_app.c

```diff
@@ -0,0 +1,1778 @@
+static void ov17_022029C8(BerryPotsAppData *data, u32 a1);
+static int ov17_02202A50(BerryPotsAppData *data);
+static void ov17_02202A84(BerryPotsAppData *data, int index);
+static void ov17_02202B58(BerryPotsAppData *data, u8 index);
+static void ov17_02202B98(BerryPotsAppData *data);
+static void ov17_02202BF8(BerryPotsAppData *data);
+static void ov17_02202C2C(void);
+static void BerryPotsApp_SetupBgConfig(BerryPotsAppData *data);
+static void BerryPotsApp_FreeBgConfig(BerryPotsAppData *data);
+static void BerryPotsApp_SetupNarc(BerryPotsAppData *data);
+static void BerryPotsApp_FreeNarc(BerryPotsAppData *data);
+static void BerryPotsApp_SetupText(BerryPotsAppData *data);
+static void BerryPotsApp_FreeText(BerryPotsAppData *data);
+static void BerryPotsApp_SetupWindows(BerryPotsAppData *data);
+static void BerryPotsApp_FreeWindows(BerryPotsAppData *data);
+static void BerryPotsApp_SetupListMenuItems(BerryPotsAppData *data);
+static void BerryPotsApp_FreeListMenuItems(BerryPotsAppData *data);
+static void BerryPotsApp_SetupSpriteRendererAndGfxHandler(BerryPotsAppData *data);
+static void BerryPotsApp_FreeSpriteRendererAndGfxHandler(BerryPotsAppData *data);
+static void BerryPotsApp_SetupSpriteSystem(BerryPotsAppData *data);
+static void BerryPotsApp_FreeSpriteSystem(BerryPotsAppData *data);
+static void BerryPotsApp_SetupSprites(BerryPotsAppData *data);
+static void BerryPotsApp_FreeSprites(BerryPotsAppData *data);
+static void ov17_02203404(BerryPotsAppData *data, int index);
+static void ov17_02203460(BerryPotsAppData *data, u8 index);
+static UnkImageStruct *ov17_022034C8(BerryPotsAppData *data, s16 x, s16 y, u8);
+static void ov17_0220351C(BerryPotsAppData *data);
+static void ov17_02203638(BerryPotsAppData *data);
+static void ov17_022036FC(BerryPotsAppData *data, u32 a1, u8 potIndex);
+static void ov17_022037C8(BerryPotsAppData *data, BOOL a1);
+static void ov17_0220387C(BerryPotsAppData *data);
+static void ov17_02203928(BerryPotsAppData *data);
+static u32 ov17_02203A34(BerryPotsAppData *data);
+static BOOL ov17_02203A54(BerryPotsAppData *data);
+static u32 ov17_02203A74(BerryPotsAppData *data, u32 a1);
+static u32 ov17_02203AD4(BerryPotsAppData *data);
+static u32 ov17_02203B68(BerryPotsAppData *data, BOOL *a1);
+static BOOL ov17_02203B88(BerryPotsAppData *data);
+static BOOL ov17_02203C20(BerryPotsAppData *data, BOOL *a1);
+static int ov17_02203C78(BerryPotsAppData *data);
+static int ov17_02203CC8(BerryPotsAppData *data);
+static u32 ov17_02203D00(BerryPotsAppData *data);
+
+void ov17_02201BC0(void) {
+    Main_SetVBlankIntrCB(NULL, NULL);
+    HBlankInterruptDisable();
+
+    GX_DisableEngineALayers();
+    GX_DisableEngineBLayers();
+
+    GX_SetVisiblePlane(0);
+    GXS_SetVisiblePlane(0);
+
+    sub_0200FBF4(0, 0);
+    sub_0200FBF4(1, 0);
+    sub_0200FBDC(0);
+    sub_0200FBDC(1);
+}
+
+BOOL BerryPotsApp_Initialize(OVY_MANAGER *manager, int *state) {
+    switch (*state) {
+        case 0: {
```

Comment:
any reason why you have the brackets for cases?

Reply:
Naming collisions and C++ habits. Fixed